src/extra/components/timepix.py

```diff
@@ -0,0 +1,612 @@
+
+from collections import defaultdict
+import re
+
+import numpy as np
+import pandas as pd
+
+from extra_data import by_id
+
+
+# Pulse indices used internally for virtual leading and trailing pulses.
+TRAILING_PULSE_INDEX = 2**31 - 1
+LEADING_PULSE_INDEX = -TRAILING_PULSE_INDEX - 1
+
+
+class Timepix3:
+    """Interface to Timepix3 event data.
+
+    Timepix3 is an event-based pixel detectors, generating a stream of
+    events of when and how long a singular pixel is detected. This
+    component gives convenient [pd.dataframe][pandas.DataFrame]-based
+    access to directly acquired raw pixel events, as well as centroided
+    events available after automatic processing. These centroids attempt
+    to group events in time and space that belong to the same particle
+    impact to a single event.
+
+    The component can be initialized with either the raw data source,
+    the virtual calibrated source containing the centroided results or
+    both. Pulse pattern information are necessary in both cases to sort
+    events into pulses, as they are originally recorded by train.
+
+    Args:
+        data (extra_data.DataCollection): Data to access Timepix3 data from.
+        detector (str, optional): Name of the detector, only
+            needed if the data includes more than one. May be any prefix
+            of either raw or centroided source name.
+        pulses (extra.components.pulses.PulsePattern, optional): Pulse
+            component to pull pulse information. If omitted, an
+            [XrayPulses][extra.components.XrayPulses] object is
+            constructed from the data.
+    """
+
+    # Only support single-chip detectors for now.
+    _instrument_re = re.compile(
+        r'^(\w{3}_\w+_\w+)\/(DET|CAL)\/\w+:daqOutput.chip0$')
```

Comment:
I assume you've tested this, but the `chip0` comes in the source name, not as a key within the schema? If so, it's a nice surprise if everything works correctly for that in EXtra-data...

Other than that, this source name is very generic. Is there no 'TIMEPIX' or similar part we can expect there?

Reply:
I might put it back, to hedge against getting any other devices with a `daqOutput.chip0` channel. But I don't feel strongly about that, and you probably have a better idea of likely patterns around this than I do.

Reply:
Coupled with the naming change, it expects the domain now end in `_TIMEPIX`. That at least applies to all known instances.

Reply:
That sounds good, thanks.